server/api/applicant.js

```diff
@@ -211,79 +216,114 @@ ORDER BY course.code
   );
 
   /**
-   * Post to magically pull forward this applicants
-   * most recent application details from previous
-   * terms
+   * Function to get the funding info for a given userid + term combo
    */
-  app.post(
-    "/api/applicant/term/new",
-    AS(async (req, res) => {
-      const userId = res.locals.userid;
-      const r = req.body;
-      // console.log(r);
-      db.tx(async (t) => {
-        // locks the users table
-        await t.oneOrNone(
-          "SELECT * FROM users WHERE username=$1 FOR NO KEY UPDATE",
-          [userId]
-        );
-        const termApps = await getAvailableApplications(req, res);
+  async function getFunding(userId, term) {
+    const dbQuery = `
+    SELECT funding FROM applicantfunding 
+    JOIN applicant ON applicantfunding.studentnum=applicant.studentnum
+    JOIN users ON applicant.id=users.id
+    WHERE users.username=$1 AND term=$2`;
```

Comment:
To make this easier to read:
```suggestion
    WHERE users.username=$1 AND applicantfunding.term=$2`;
```